src/layer.tsx

```diff
@@ -329,3 +329,9 @@ export default class Layer extends React.Component<Props> {
     return null;
   }
 }
+
+const LayerWithMap: React.SFC<Props> = props => (
+  <MapConsumer>{({ map }) => <Layer {...props} map={map} />}</MapConsumer>
```

Comment:
I might be missing something, but this could use `withMapContext`, right? A bunch of other places, too

Reply:
Good question: I added `withMapContext` purely to get `layer-events-hoc.tsx` to work with the new context API. I would discourage using it anywhere else and use the `<MapConsumer>` _function as a child_ pattern instead... Unless you really need to create a side-effect in a react lifecycle hook (like `layer-events-hoc.tsx` does) `withMapContext` gives you an escape hatch.